### DISCUSSION TITLE
Hypothetically speaking, Is it possible to edit harlequin to be a general IDE?

### DISCUSSION THREAD

**Discussion by agoodfellow123** [tconbeer/harlequin#416] [General]
Hello, I have a small brain and big dreams. I noticed there are no IDEs for terminals with LSP support and keybindings that are made for humans (I'm looking at you Vim). The closest thing we have is Micro but it's more of a text editor with LSP plugin than an IDE. I looked at the source code for an hour and since I still can not grasp the concept of oop I got lost in the first 3 minutes. What I'm trying to achieve is not that simple but if anyone wants to do this kind of project I can help with my beginner knowledge.

Things we should change:
The bottom split must be a terminal.
We should add a LSP syntax support and be able to use other LSP's.
Feel free to add more...

**Comment by tconbeer** [comment]
No, not really. You would want to start over instead, maybe using Harlequin's text editor widget, which can be found [here](https://github.com/tconbeer/textual-textarea).

But really I think you're just looking for [modeless vim](https://github.com/SebastianMuskalla/ModelessVim) or emacs?

**Reply by agoodfellow123** [reply]
I see, thank you for your answer. I tried both novim (for neovim) and ergoemacs (for emacs), both are hacks around the problem and some features gets broken with those. That's why I wanted to use this project as a template.
